### PR TITLE
Using fully qualified names in library Transformations as identifiers

### PR DESCRIPTION
**Pull request by kuzdogan**
See #1486 

## Summary by CodeRabbit

- **New Features**
	- Introduced new properties for handling link references, enhancing verification capabilities.
	- Added the `LinkReferences` interface for structured representation of library paths in the codebase.
	- Implemented new libraries for key and claim management, improving decentralized application functionality.
	- Included a comprehensive ABI and metadata for a new smart contract for claims management.

- **Bug Fixes**
	- Enhanced error handling for library placeholder mismatches in bytecode verification.

- **Tests**
	- Expanded testing suite for contracts compiled with older Solidity versions, enhancing verification robustness.

- **Chores**
	- Added a new dependency to improve semantic versioning management within the project.

### CODE REVIEW

File: services/server/src/server/services/storageServices/SourcifyDatabaseService.ts [hunk 1]
```diff
@@ -307,9 +309,24 @@ export class SourcifyDatabaseService
...
         sourcifyMatch.runtime_values.libraries,
-      );
+      )) {
+        let formattedKey;
+        // Solidity >= 0.5.0 is __$keccak256(file.sol:MyLib)$__ (total 40 characters)
+        if (semver.gte(sourcifyMatch.metadata.compiler.version, "0.5.0")) {
+          formattedKey =
+            "__$" + keccak256Str(key).slice(2).slice(0, 34) + "$__";
+        } else {
+          // Solidity < 0.5.0 is __MyLib__________ (total 40 characters)
```
Comment (manuelwedler) [inline services/server/src/server/services/storageServices/SourcifyDatabaseService.ts:323]: Can we add a test for the case of Solidity < 0.5.0?

**Review by manuelwedler — COMMENTED** [review]
I would add another test, other than that it looks good

**Comment by kuzdogan** [comment]
Thanks for pushing for that @manuelwedler. Added a test, and found two bugs during that:
1. It wouldn't work as expected if the library FQN was longer
2. We were saving the `callProtection` transformation with `offset: 0` whereas the first byte `0x73` is not changed. Only the next 20 bytes (contract's own address) is replaced.

**Review by manuelwedler — APPROVED** [review]
Nice, I'm glad you found these problems :)

**Comment by kuzdogan** [comment]
@coderabbitai review

File: services/server/src/server/services/storageServices/SourcifyDatabaseService.ts [hunk 2]
```diff
@@ -307,9 +309,25 @@
...
+          const libName = key.split(":")[1];
+          const trimmedLibName = libName.slice(0, 36); // in case it's longer
+          formattedKey = "__" + trimmedLibName.padEnd(38, "_");
+        }
```
[the remark this answers is not in the document]
Comment (kuzdogan) [reply services/server/src/server/services/storageServices/SourcifyDatabaseService.ts:327]: Generate unit testing code

[the remark this answers is not in the document]
Comment (kuzdogan) [reply services/server/src/server/services/storageServices/SourcifyDatabaseService.ts:327]: Do it in nock, chai and mocha

**Comment by Chator1** [comment]
0xFD689e5f2d8d9Aec0aD328225Ae62FdBDdb30328

**Pull request merged**